lib/Service/CircleHelper.php

```diff
@@ -260,6 +260,37 @@ public function getLevel(string $circleId, string $userId): int {
 		return $member->getLevel();
 	}
 
+	/**
+	 * @throws NotFoundException
+	 * @throws NotPermittedException
+	 * @throws MissingDependencyException
+	 */
+	public function canLeave(string $circleId, string $userId): bool {
+		if (is_null($this->circlesManager)) {
+			throw new MissingDependencyException($this->dependencyInjectionError);
+		}
+
+		try {
+			$this->startSession($userId);
```

Comment:
not sure what it means, gives me shivers

Reply:
Hehe, I understand. That's how the Circles API works. It starts a session for `CirclesManager` as the user. Afterwards, we can probe the `CirclesManager` for circles in the name of this user. See https://github.com/nextcloud/circles/blob/3bf2a7bd617785d94c382222d52e96814d2569f3/lib/CirclesManager.php#L175